src/lib/runVPAID.js

```diff
@@ -1,18 +1,18 @@
+import { DATA_READY, SUPPORTED_VPAID_VERSION, VPAID_STARTED } from '../constants';
+
 // See https://marcj.github.io/css-element-queries/ for ResizeSensor docs
 import ResizeSensor from 'css-element-queries/src/ResizeSensor';
 import { VASTTracker } from 'vast-client';
-import loadVPAID from '../helpers/loadVPAID';
-import createNode from '../helpers/createNode';
-import isVPAIDVersionSupported from '../helpers/isVPAIDVersionSupported';
 import VPAIDWrapper from '../lib/VPAIDWrapper';
-import { DATA_READY, SUPPORTED_VPAID_VERSION, VPAID_STARTED } from '../constants';
+import isVPAIDVersionSupported from '../helpers/isVPAIDVersionSupported';
+import loadVPAID from '../helpers/loadVPAID';
 
 const viewMode = 'normal';
 let originalDimensions;
 
 export default async function runVPAID(creative, VPAIDSource, vastClient, ad) {
     const vastTracker = new VASTTracker(vastClient, ad, creative);
-    const { adParameters } = creative;
+    const adParameters = creative.variations[0].adParameters;
```

Comment:
Wouldn't this use the Non-Linear adParameters always?

Reply:
I just checked with this [linear tag](https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/single_ad_samples&ciu_szs=300x250&impl=s&gdfp_req=1&env=vp&output=vast&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ct%3Dlinearvpaid2js&correlator=), and it does fail. This needs to be validated when using Linear vs Non-Linear.
```
error - Instance videoContainer2 log: TypeError: Cannot read property '0' of undefined
    at e.<anonymous> (runVPAID.js:15)
    at c (runtime.js:45)
    at Generator._invoke (runtime.js:274)
    at Generator.forEach.e.<computed> [as next] (runtime.js:97)
    at h (asyncToGenerator.js:3)
    at s (asyncToGenerator.js:25)
    at asyncToGenerator.js:32
    at new Promise (<anonymous>)
    at e.<anonymous> (asyncToGenerator.js:21)
    at e.qe (runVPAID.js:39)```

Reply:
Rightly so, fixed :)